### PR TITLE
Task 3

### PR DESCRIPTION
**Pull request by sergeypogorelov**
To check the movies search page you need to comment it out in app.component.tsx

### CODE REVIEW

File: src/client/app/app.component.tsx [hunk 1]
```diff
@@ -1,5 +1,16 @@
...
 
-import { AppProps } from './app-props.interface';
+import { Page } from './layout/page.component';
+import { SearchMoviePage } from './pages/search-movie/search-movie-page.component';
+import { ViewMoviePage } from './pages/view-movie/view-movie-page.component';
 
-export const App = (props: AppProps) => <h1>Hello {props.name}!</h1>;
+export const App = () => {
+  return (
+    <>
```
Comment (IrinaShilina) [inline src/client/app/app.component.tsx:9]: You don't really need a fragment here. You already have wrapping `Page` component.

File: src/client/app/layout/footer/footer.component.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import './footer.component.scss';
+
+import React, { FunctionComponent } from 'react';
+
+import { TitleMain } from '../../shared/title-main/title-main.component';
```
Comment (IrinaShilina) [inline src/client/app/layout/footer/footer.component.tsx:5]: Minor: we don't usually include `.component` in the file name in React projects. I assume it's a matter of taste but keep in mind that it's not a common approach.

File: src/client/app/layout/page.component.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React, { FunctionComponent } from 'react';
+
+import { Footer } from './footer/footer.component';
+
+export const Page: FunctionComponent<{}> = (props) => {
+  return (
+    <div className="container">
+      {props.children}
```
Comment (IrinaShilina) [inline src/client/app/layout/page.component.tsx:8]: You can try the object destructuring.

```
export const Page: FunctionComponent<{}> = ({ children }) => {
  return (
    <div className="container">
      {children}
```

File: src/client/app/pages/search-movie/movie-search/movie-search.component.tsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import './movie-search.component.scss';
+
+import React, { FunctionComponent } from 'react';
+
+export const MovieSearch: FunctionComponent<{}> = (props) => {
+  return (
+    <form className="movie-search">
+      <input className="movie-search-control" />
```
Comment (IrinaShilina) [inline src/client/app/pages/search-movie/movie-search/movie-search.component.tsx:8]: Minor: all inputs should have labels.

File: src/client/app/pages/search-movie/search-movie-page.component.scss [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+.search-movie-page {
+  .header-title {
+    margin-bottom: 35px;
+  }
+  .header-content {
+    .movie-search {
+      margin-bottom: 15px;
+    }
+    margin-left: 60px;
```
Comment (IrinaShilina) [inline src/client/app/pages/search-movie/search-movie-page.component.scss:11]: It's cleaner to add nested rules after common rules. Also, consider adding Prettier in order to maintain easy-to-read spacing. 

E.g.

```
  .header-content {
    margin-left: 60px;

    .movie-search {
      margin-bottom: 15px;
    }
}
```

File: src/client/app/shared/error-boundary/error-boundary.component.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  };
+
+  static getDerivedStateFromError(error: any): ErrorBoundaryState {
+    return {
+      hasError: true,
+    };
+  }
+
+  componentDidCatch(error: any, errorInfo: any) {
+    console.log(error, errorInfo);
```
Comment (IrinaShilina) [inline src/client/app/shared/error-boundary/error-boundary.component.tsx:19]: How about `console.error()`?

File: src/client/app/shared/error-boundary/error-boundary.component.tsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+      return (
+        <ModalAlert title="Error" onClose={this.handleOnModalClose}>
+          <p className="h5 text-center m-0">
+            Something went wrong. Click OK to reload the page.
+          </p>
+        </ModalAlert>
+      );
+    }
+
+    return this.props.children;
```
Comment (IrinaShilina) [inline src/client/app/shared/error-boundary/error-boundary.component.tsx:33]: You can write this simpler:
```
return this.state.hasError ? (
        <ModalAlert title="Error" onClose={this.handleOnModalClose}>
          <p className="h5 text-center m-0">
            Something went wrong. Click OK to reload the page.
          </p>
        </ModalAlert>
) 
  : 
this.props.children
```

File: src/client/app/shared/movies-list/movies-list.component.tsx [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+
+import { MovieItem } from './movie-item/movie-item.component';
+
+export const MoviesList: FunctionComponent<{}> = (props) => {
+  return (
+    <div className="movies-list">
+      <div className="row">
+        <div className="col-4 movie-wrapper">
+          <MovieItem />
+        </div>
```
Comment (IrinaShilina) [inline src/client/app/shared/movies-list/movies-list.component.tsx:13]: I suggest to add `map()` function.

```
const movies = [1,2,3,4,5,6]
<div className="row">
 {movies.map(key=> (
    <div className="col-4 movie-wrapper" key={key}>
       <MovieItem />
     </div>
))}

```

File: src/client/app/layout/page.component.tsx (hunk 3 above)
Comment (sergeypogorelov) [reply src/client/app/layout/page.component.tsx:8]: Makes sense, thanks

File: src/client/app/shared/error-boundary/error-boundary.component.tsx (hunk 6 above)
Comment (sergeypogorelov) [reply src/client/app/shared/error-boundary/error-boundary.component.tsx:19]: Fixed

File: src/client/app/shared/error-boundary/error-boundary.component.tsx (hunk 7 above)
Comment (sergeypogorelov) [reply src/client/app/shared/error-boundary/error-boundary.component.tsx:33]: Fixed

File: src/client/app/shared/movies-list/movies-list.component.tsx (hunk 8 above)
Comment (sergeypogorelov) [reply src/client/app/shared/movies-list/movies-list.component.tsx:13]: Good point, thanks

File: src/client/app/pages/search-movie/movie-search/movie-search.component.tsx (hunk 4 above)
Comment (sergeypogorelov) [reply src/client/app/pages/search-movie/movie-search/movie-search.component.tsx]: Fixed

**Pull request merged**